packages/opencensus-nodejs/test/test-tracing.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as assert from 'assert';
+
+import {defaultConfig} from '../src/trace/config/default-config';
+import {Constants} from '../src/trace/constants';
+import {Tracing} from '../src/trace/tracing';
+
+
+
+const NOOP_EXPORTER = new classes.NoopExporter();
+describe('Tracing', () => {
+  Constants.DEFAULT_INSTRUMENTATION_MODULES = ['http', 'https'];
+
+  /** Should create a Tracing instance */
+  describe('new Tracing()', () => {
+    it('should create a Tracer instance', () => {
+      const tracing = new Tracing();
+      assert.ok(tracing instanceof Tracing);
+    });
+  });
+
+  /** Should get the singleton trancing instance. */
+  describe('static get instance()', () => {
+    it('should get the singleton trancing instance', () => {
+      const tracing = Tracing.instance;
+      assert.ok(tracing instanceof Tracing);
+    });
+  });
+
+  /** Should return a started tracing instance */
+  describe('start()', () => {
+    let aTracing: types.Tracing;
+    const tracing = new Tracing();
+    // tslint:disable:no-any
+    beforeEach(() => {
+      if (tracing.active) {
+        tracing.stop();
+      }
+    });
+
+    it('should return a tracing instance', () => {
+      aTracing = tracing.start();
+      assert.ok(aTracing instanceof Tracing);
+      assert.ok(tracing.active);
+      assert.ok(aTracing.tracer.active);
+    });
+
+    it('should start with default config', () => {
```

Comment:
Currently this test seems to need to change every time a new configuration field is added. Is there a way to automatically do this?